corehq/apps/integration/kyc/models.py

```diff
@@ -116,7 +125,115 @@ def get_user_objects_by_ids(self, obj_ids):
             UserDataStore.CUSTOM_USER_DATA,
             UserDataStore.USER_CASE,
         ):
-            return [CommCareUser.get_by_user_id(id_) for id_ in obj_ids]
+            return KycUser.from_hq_user_or_case_objects(
+                self,
+                [CommCareUser.get_by_user_id(id_) for id_ in obj_ids]
+            )
         elif self.user_data_store == UserDataStore.OTHER_CASE_TYPE:
             assert self.other_case_type
-            return CommCareCase.objects.get_cases(obj_ids, self.domain)
+            return KycUser.from_hq_user_or_case_objects(self, CommCareCase.objects.get_cases(obj_ids, self.domain))
+
+
+class KycUser:
+
+    def __init__(self, kyc_config, user_or_case_obj):
+        """
+        :param kyc_config: kyc configuration for the domain
+        :param user_or_case_obj: can be an instance of 'CommcareUser' or 'CommcareCase' based on the configuration.
+        """
+        self.kyc_config = kyc_config
+        self.user_or_case_obj = user_or_case_obj
+
+    @cached_property
+    def user_id(self):
+        if isinstance(self.user_or_case_obj, CommCareUser):
+            return self.user_or_case_obj.user_id
+        return self.user_or_case_obj.case_id
+
+    @cached_property
+    def user_data(self):
+        if self.kyc_config.user_data_store == UserDataStore.CUSTOM_USER_DATA:
+            return self.user_or_case_obj.get_user_data(self.kyc_config.domain).to_dict()
+        elif self.kyc_config.user_data_store == UserDataStore.USER_CASE:
+            custom_user_case = self.user_or_case_obj.get_usercase()
+            if not custom_user_case:
+                raise UserCaseNotFound("User case not found for the user.")
+            return custom_user_case.case_json
+        # UserDataStore.OTHER_CASE_TYPE
+        return self.user_or_case_obj.case_json
+
+    @property
+    def kyc_last_verified_at(self):
+        return self.user_data.get('kyc_last_verified_at')
+
+    @property
+    def kyc_is_verified(self):
+        value = self.user_data.get('kyc_is_verified')
+        if value:
+            # convert to boolean from string
+            value = ast.literal_eval(value)
+        elif value == '':  # for user custom data when field is defined as a custom field
+            value = None
+        return value
```

Comment:
`ast.literal_eval()` is safe, but still more expensive than we need because we know there are only four possible values of `value`. I'd replace it with two `if` statements:
```suggestion
    @property
    def kyc_is_verified(self):
        value = self.user_data.get('kyc_is_verified')
        # value can be '' when field is defined as a custom field in custom user data
        assert value in (None, 'True', 'False', '')
        if value == 'True':
            return True
        if value == 'False':
            return False
        return None
```

Reply:
Addressed in fb71d76fc3043486d2fed17947ca772f89e92d34